packages/myst-cli/src/project/fromTOC.ts

```diff
@@ -31,156 +32,44 @@ import { globSync } from 'glob';
 import { isDirectory } from 'myst-cli-utils';
 
 export const DEFAULT_INDEX_FILENAMES = ['index', 'readme', 'main'];
-
-const DEFAULT_INDEX_WITH_EXT = ['.md', '.ipynb']
+const DEFAULT_INDEX_WITH_EXT = ['.md', '.ipynb', '.myst.json']
   .map((ext) => DEFAULT_INDEX_FILENAMES.map((file) => `${file}${ext}`))
   .flat();
 
 type EntryWithoutPattern = FileEntry | URLEntry | FileParentEntry | URLParentEntry | ParentEntry;
 
-/** Sort any folders/files to ensure that `chapter10`, etc., comes after `chapter9` */
-export function sortByNumber(a: string, b: string) {
-  // Replace any repeated numbers with padded zeros
-  const regex = /(\d+)/g;
-  const paddedA = a.replace(regex, (match) => match.padStart(10, '0'));
-  const paddedB = b.replace(regex, (match) => match.padStart(10, '0'));
-  // Compare the modified strings
-  return paddedA.localeCompare(paddedB);
-}
-
-/**
- * Take path parts and fill into nested object structure.
- *
- * For example, given:
- *
- * parts = ['one', 'two', 'three']
- *
- * This builds:
- *
- * {
- *   one: {
- *     two: {
- *       three: {},
- *     },
- *   },
- * }
- */
-export function objFromPathParts(parts: string[], obj?: Record<string, any>) {
-  if (!obj) obj = {};
-  const [part, ...remainingParts] = parts;
-  if (!part) return obj;
-  if (!obj[part]) obj[part] = {} as Record<string, any>;
-  obj[part] = objFromPathParts(remainingParts, obj[part]);
-  return obj;
-}
+export function comparePaths(a: string, b: string): number {
```

Comment:
@fwkoch this function now performs the directory-sensitive string comparison:

1. Paths with non-matching directories are nat-sorted
2. Paths with matching directories are nat-sorted if neither file is an index, or both files are index files
3. Paths with non-matching directories are sorted in preference of the index file if one file is an index

Does this match your expectations?